pennylane/operation.py

```diff
@@ -712,6 +712,7 @@ def __init__(self, *params, wires=None, do_queue=True, id=None):
         self._name = self.__class__.__name__  #: str: name of the operator
         self._id = id
         self.queue_idx = None  #: int, None: index of the Operator in the circuit queue, or None if not in a queue
+        self._pauli_rep = None
```

Comment:
I believe using a `_pauli_rep` property would make more sense. This way users can't modify its value.

Reply:
I decided not to use a property because we want this quantity to be resolved on construction. That way, if it is queried for a complicated nested arithmetic operator, we save computation time for its construction.

Reply:
We might consider adding a public `op.pauli_rep` property in the future, but for now, I think it's better to leave it as a private variable.  That way we can make adjustments internally as we see fit.

But it would be good to add a "docstring" comment here:

```suggestion
        self._pauli_rep = None # Union[PauliSentence, None]: Representation of the operator as a pauli sentence, if applicable
```

Or something like that.